pkg/apis/apps/validation/validation.go

```diff
@@ -69,6 +70,13 @@ func ValidatePodTemplateSpecForStatefulSet(template *api.PodTemplateSpec, select
 	return allErrs
 }
 
+var supportedPersistentVolumeClaimDeletePolicy = sets.NewString(
```

Comment:
Oh one more thing, if you want to make the field mutable, you'll need to update https://github.com/kubernetes/kubernetes/blob/981003ad6c31e8631583aba57848b641a58246a6/pkg/apis/apps/validation/validation.go#L164

Reply:
Good point, thanks, done.